lib/engine/game/g_18_carolinas.rb

```diff
@@ -7,6 +7,37 @@ module Engine
   module Game
     class G18Carolinas < Base
       load_from_json(Config::Game::G18Carolinas::JSON)
+
+      def init_companies(players)
+        super.take(players.size)
+      end
+
+      def setup
+        @companies.each_with_index do |company, index|
+          price = company.value
+          player = @players[index]
+          company.owner = player
+          @players[index].companies << company
```

Comment:
this these two lines to player instead of @players[index]

Reply:
Fixed, although I don't understand why this works. I thought player was a temporary variable and would not spend the money of give the companies. How do these changes get back into @players?